### PR TITLE
ENH : reduced constraints on ReadOnlyControllers/Collections

### PR DESCRIPTION
**Pull request by nfaugout**
You can now have interfaces or base class in read only controllers.

### CODE REVIEW

**Comment by nfaugout** [comment]
@srabere this PR correspond to your need in Timmi to have an API of something thats not really an EntityBase, thus something that can be an aggregate of other objects or an interface implemented by multiple concrete classes.

File: Web/RDD.Web.Tests/Models/User.cs [hunk 1]
```diff
@@ -1,10 +1,16 @@
-﻿using RDD.Domain.Models;
+﻿using RDD.Domain;
+using RDD.Domain.Models;
 
 namespace RDD.Web.Tests.Models
 {
-    public class User : EntityBase<User, int>
+    public class User : EntityBase<User, int>, IEntityBase<IUser, int>, IUser
```
Comment (LuccaIntegration) [inline Web/RDD.Web.Tests/Models/User.cs]: MINOR 'IUser' implements 'IEntityBase<IUser, int>' so 'IEntityBase<IUser, int>' can be removed from the inheritance list. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1939)

File: Web/RDD.Web.Tests/Models/IUserWebController.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using RDD.Web.Controllers;
+using RDD.Web.Helpers;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RDD.Web.Tests.Models
+{
+    public class IUserWebController : ReadOnlyWebController<IUser, int>
```
Comment (LuccaIntegration) [inline Web/RDD.Web.Tests/Models/IUserWebController.cs]: MINOR Rename class 'IUserWebController' to match camel case naming rules, consider using 'UserWebController'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS101)

**Comment by LuccaIntegration** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Web/RDD.Web.Tests/Models/IUserWebController.cs (hunk 2 above)
Comment (nfaugout) [reply Web/RDD.Web.Tests/Models/IUserWebController.cs]: Celui là je le laisse, c'est fait pour dire qu'on a affaire à une interface

**Comment by LuccaIntegration** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Web/RDD.Web.Tests/Models/IUserWebController.cs (hunk 2 above)
Comment (rducom) [reply Web/RDD.Web.Tests/Models/IUserWebController.cs]: I agree with sonar ;)

**Comment by nfaugout** [comment]
I could drop it on RestCollection (not only ReadOnly ones), and throw an Exception on the generic implementation of InstanciateEntity( ).

This would force everyone to never use IRestCollection<T>, but instead custom overrides implementing IMyEntitiesCollection for instance.

This would introduce a breaking change though, but for the good sake ;)

**Comment by nfaugout** [comment]
@Poltuu same here

**Pull request merged**